pkg/metrics/metrics.go

```diff
@@ -33,8 +33,32 @@ var (
 		Name: "mapi_mao_collector_up",
 		Help: "Machine API Operator metrics are being collected and reported successfully",
 	}, []string{"kind"})
+
+	// FailedMachineSetProvisionCount calculates the number of failures for a specific machine
+	FailedMachineSetProvisionCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "failed_machine_sets_total",
+			Help: "Number of times machine set provisioning has failed.",
+		}, []string{"name", "namespace", "timestamp", "reason"},
```

Comment:
I'm not sure we should include the timestamp, does it add any extra information for the user do you think?

Reply:
Does not look like, if only having additional unique key for tracking successes or failures for same named machine sets, which sounds too much.

Reply:
Prometheus will allow you to see this anyway, as it will track the value over time. So you can use a query to work out when the value changed and get a spike on the graph when it was incremented. Good thought though! Definitely the right line of thinking